tests/components/eafm/test_sensor.py

```diff
@@ -0,0 +1,431 @@
+"""Tests for polling measures."""
+import datetime
+
+import aiohttp
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+DUMMY_REQUEST_INFO = aiohttp.client.RequestInfo(
+    url="http://example.com", method="GET", headers={}, real_url="http://example.com"
+)
+
+CONNECTION_EXCEPTIONS = [
+    aiohttp.ClientConnectionError("Mock connection error"),
+    aiohttp.ClientResponseError(DUMMY_REQUEST_INFO, [], message="Mock response error"),
+]
+
+
+async def async_setup_test_fixture(hass, mock_get_station, initial_value):
+    """Create a dummy config entry for testing polling."""
+    mock_get_station.return_value = initial_value
+
+    entry = MockConfigEntry(
+        version=1,
+        domain="eafm",
+        entry_id="VikingRecorder1234",
+        data={"station": "L1234"},
+        title="Viking Recorder",
+        connection_class=config_entries.CONN_CLASS_CLOUD_PUSH,
+    )
+    entry.add_to_hass(hass)
+
+    assert await async_setup_component(hass, "eafm", {})
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)